lnwallet/fee_estimator.go

```diff
@@ -8,19 +8,68 @@ import (
 	"github.com/roasbeef/btcutil"
 )
 
+// SatPerVByte represents a fee rate in satoshis per vbyte.
+type SatPerVByte btcutil.Amount
+
+// FeePerWeight converts the fee rate into SatPerWeight.
+func (s SatPerVByte) FeePerWeight() SatPerWeight {
+	return s.FeePerKWeight().FeePerWeight()
+}
+
+// FeePerKWeight converts the fee rate into SatPerKWeight.
+func (s SatPerVByte) FeePerKWeight() SatPerKWeight {
+	return SatPerKWeight(s * 1000 / blockchain.WitnessScaleFactor)
+}
+
+// SatPerWeight represents a fee rate in satoshis per weight unit.
+type SatPerWeight btcutil.Amount
+
+// FeePerVSize converts the fee rate into SatPerVByte.
+func (s SatPerWeight) FeePerVSize() SatPerVByte {
+	return s.FeePerKWeight().FeePerVSize()
+}
+
+// FeePerKWeight converts the fee rate into SatPerKWeight.
+func (s SatPerWeight) FeePerKWeight() SatPerKWeight {
+	return SatPerKWeight(s * 1000)
+}
+
+// TODO: call Kw our Kwu?
+// SatPerKWeight represents a fee rate in satoshis per kilo weight unit.
+type SatPerKWeight btcutil.Amount
+
+// FeePerWeight converts the fee rate into SatPerWeight.
+func (s SatPerKWeight) FeePerWeight() SatPerWeight {
+	f := SatPerWeight(s / 1000)
+
+	// If the fee rate is non-zero, then we should
+	// always return at least 1.
+	if s > 0 && f < 1 {
```

Comment:
TODO: I realized this can be solved by just always rounding up. Will look into if that takes care of all these cases.